src/components/src/map/map-control-panel.tsx

```diff
@@ -97,41 +120,76 @@ function MapControlPanelFactory() {
       onClick,
       scale = 1,
       isExport,
-      logoComponent
-    }) => (
-      <StyledMapControlPanel
-        className="map-control-panel"
-        style={{
-          transform: `scale(${scale})`,
-          marginBottom: '8px !important'
-        }}
-      >
-        <StyledMapControlPanelHeader>
-          {isExport && logoComponent ? (
-            logoComponent
-          ) : header ? (
-            <span style={{verticalAlign: 'middle'}}>
-              <FormattedMessage id={header} />
-            </span>
+      logoComponent,
+      mapState,
+      onToggleSplitMapViewport
+    }) => {
+      const {isViewportSynced, isZoomLocked} = mapState || {};
+      const onUnlockViewportChange = useCallback(() => {
+        onToggleSplitMapViewport?.({isViewportSynced: !isViewportSynced});
+      }, [isViewportSynced, onToggleSplitMapViewport]);
+
+      const onSyncZoomChange = useCallback(() => {
+        onToggleSplitMapViewport?.({isZoomLocked: !isZoomLocked});
+      }, [isZoomLocked, onToggleSplitMapViewport]);
+
+      return (
+        <StyledMapControlPanel
+          className="map-control-panel"
+          style={{
+            transform: `scale(${scale})`,
+            marginBottom: '8px !important'
+          }}
+        >
+          {mapState?.isSplit ? (
+            <StyledMapControlPanelHeaderSplitViewportsTools>
```

Comment:
Thanks for bringing this one in, @igorDykhta. When I test the deployed PR demo or run it locally myself, I can't get these conditional controls to render. If I force this to `true` it would look like this in a split maps view:

![Screenshot 2023-03-26 at 10 46 33 PM](https://user-images.githubusercontent.com/4933392/227828340-0f66085b-4cbb-440a-a911-77541b1cdbf1.png)

I'm not sure why yet, but in the demo app code it seems like the parents of this component might now be passing around props that are a flattened `mapState` instead of when this was originally written to pass in the full `mapState`. 🤔 Either way, `mapState` seems to evaluate to `undefined` now when I try to debug.

Reply:
@jwasilgeo should be fixed